rayon-core/src/scope/mod.rs

```diff
@@ -262,22 +277,30 @@ where
     R: Send,
 {
     in_worker(|owner_thread, _| {
-        unsafe {
-            let scope: Scope<'scope> = Scope {
-                owner_thread_index: owner_thread.index(),
-                registry: owner_thread.registry().clone(),
-                panic: AtomicPtr::new(ptr::null_mut()),
-                job_completed_latch: CountLatch::new(),
-                marker: PhantomData,
-            };
-            let result = scope.execute_job_closure(op);
-            scope.steal_till_jobs_complete(owner_thread);
-            result.unwrap() // only None if `op` panicked, and that would have been propagated
-        }
+        let scope = Scope::<'scope>::new(owner_thread);
+        unsafe { scope.base.complete(owner_thread, || op(&scope)) }
+    })
+}
+
+/// TODO: like `scope()` but FIFO
```

Comment:
Still needs docs, but I guess you know that =) 

(This comment is more for me)